.github/workflows/publish-release.yml

```diff
@@ -145,6 +153,7 @@ jobs:
           echo "Check Skipped, Mark Green for Pipeline Execution"
 
   check-upgrade-handler-updated:
+    if: ${{ github.event.inputs.skip_checks != 'true' && (startsWith(github.ref, 'refs/heads/release/v') || startsWith(github.ref, 'refs/heads/hotfix/v')) }}
```

Comment:
Instead of putting this logic into every single check you can add a job that runs at the start to check_branch_name and then fail if it doesn't match.